service/controller/resource/azureclusterconditions/conditionready.go

```diff
@@ -0,0 +1,55 @@
+package azureclusterconditions
+
+import (
+	"context"
+
+	azureconditions "github.com/giantswarm/apiextensions/v3/pkg/conditions/azure"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	capz "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiconditions "sigs.k8s.io/cluster-api/util/conditions"
+)
+
+func (r *Resource) ensureReadyCondition(ctx context.Context, azureCluster *capz.AzureCluster) error {
+	r.logDebug(ctx, "setting condition Ready")
+
+	// Note: This is an incomplete implementation that checks only resource
```

Comment:
Is the final implementation still TODO?

Reply:
yes

Reply:
But this is good enough as is for what we needed for node pools release.

Reply:
I will continue covering more resources with conditions, so it's more precise, but we should not have any issues for Creating and Upgrading that were needed for the front end to show the correct cluster state.